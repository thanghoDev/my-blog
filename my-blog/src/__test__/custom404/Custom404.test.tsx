import React from 'react';
import { render, screen } from '@testing-library/react';

import Custom404 from '@/pages/404';

describe('testing custom 404 page', () => {
  test('render custom 404 page', () => {
    const { container } = render(<Custom404 />);
    expect(container).toMatchSnapshot();
    const heading = screen.getByText(/Oooops.../i);
    expect(heading).toBeInTheDocument();
    const title = screen.getByText(/that page cannot be found/i);
    expect(title).toBeInTheDocument();
  });
});
