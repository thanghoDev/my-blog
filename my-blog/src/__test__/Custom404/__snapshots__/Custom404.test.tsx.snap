// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing custom 404 page render custom 404 page 1`] = `
<div>
  <div
    class="text-center pt-5 pb-5 vh-75"
  >
    <h1>
      Oooops...
    </h1>
    <h2
      class="mt-5 mb-5"
    >
      That page cannot be found
    </h2>
    <p>
      Go back to the
       
      <a
        class="text-decoration-none text-secondary"
        href="/"
      >
        Homepage
      </a>
    </p>
  </div>
</div>
`;
